modules/imgproc/src/filter.simd.hpp

```diff
@@ -3288,6 +3288,9 @@ Ptr<BaseFilter> getLinearFilter(
     if( sdepth == CV_32F && ddepth == CV_32F )
         return makePtr<Filter2D<float, Cast<float, float>, FilterVec_32f> >
             (kernel, anchor, delta, Cast<float, float>(), FilterVec_32f(kernel, 0, delta));
+    if( sdepth == CV_32F && ddepth == CV_64F )
+        return makePtr<Filter2D<float,
+            Cast<double, double>, FilterNoVec> >(kernel, anchor, delta);
```

Comment:
OpenCV library should stay efficient.
Current binary code is really large. We should not explode it by adding of extra variants.

OpenCV has 7+ basic types. So we still need to have 7 variants of implementations.
The most part of operations doesn't allow to mix input/output types - this is done for a reason.
Support of mixed types would make problem with binary size much worse:
- 2 inputs => 7*7 = 49 variants
- 3 different inputs => 343 variants

So, there is a policy to avoid processing of mixed types. Or process them through conversion to common "working" type.
If user want double accuracy, then convert inputs to doubles first.

I believe we should fix the documentation instead.

/cc @vpisarev @asmorkalov